### PR TITLE
DDPB-4135: Update Report entity type constants to meaningful names

### PR DESCRIPTION
**Pull request by acsauk**
## Purpose
We refer to the type of report in the codebase using its numerical format. While it's possible to refer to confluence pages and code comments it's not always immediately clear if the report type you are using is correct for the deputy type and client circumstances. This update gives each report type const in `Report` a deputy and form type name rather than just reflecting the numerical representation of this. E.g. :

```php
// Not helpful
const TYPE_104 = '104';

// More helpful
const LAY_HW_REPORT_TYPE = '104';
```

I've also grouped various common report types together so we can refer to them in code. For example, you can now check if a report type belongs to a certain user group or has certain sections by checking if it appears in one of the grouping arrays:

```php
in_array('104', Report::allRolesHwAndCombinedReportTypes()) // true
```

Fixes DDPB-4135

## Checklist
- [x] I have performed a self-review of my own code
- [ ] I have updated documentation (Confluence/ADR/tech debt doc) where relevant
- [ ] I have added tests to prove my work
- [ ] The product team have approved these changes

### CODE REVIEW

File: api/src/Entity/Report/Report.php [hunk 1]
```diff
@@ -1431,4 +1400,81 @@ public function setBenefitsSectionReleaseDate(?DateTime $benefitsSectionReleaseD
...
+    }
+
+    public static function profPfaAndCombinedReportTypes(): array
+    {
+        return [
+            self::PROF_PFA_LOW_ASSETS_TYPE, self::PROF_PFA_HIGH_ASSETS_TYPE, self::PROF_COMBINED_LOW_ASSETS, self::PROF_COMBINED_HIGH_ASSETS,
+        ];
+    }
+
+    public static function allRolesLowAssetsReportTypes(): array
```
Comment (Gugandeep) [inline api/src/Entity/Report/Report.php:1477]: Is this function needed/used for anything?

**Review by Gugandeep — COMMENTED** [review]
Just a minor comment, looks like an unused/unfinished function.

But besides that, looks good.
Really nice now that you can see a report type and know what's going to be in it!

**Review by coderste — APPROVED** [review]
LGTM 👍🏻

File: api/src/Entity/Report/Report.php (hunk 1 above)
Comment (acsauk) [reply api/src/Entity/Report/Report.php:1477]: Good spot - think I started this and decided we didn't need it. I'll remove.

**Review by coderste — DISMISSED** [review]

**Review by coderste — APPROVED** [review]

**Pull request merged**